src/utils/database-common/src/transactions/db_transaction_manager.rs

```diff
@@ -102,8 +135,9 @@ impl TransactionRef {
     pub fn into_maybe_transaction<DB: sqlx::Database>(
         self,
     ) -> Option<sqlx::Transaction<'static, DB>> {
-        let m = Arc::try_unwrap(self.inner).unwrap().into_inner();
-        m.maybe_transaction
+        Arc::try_unwrap(self.inner)
+            .ok()
+            .and_then(|m| m.into_inner().maybe_transaction)
```

Comment:
This seems very dangerous - we `unwrap()` the result of extracting from `Arc` to ensure that `into_maybe_transaction` is called ONLY when there are no components holding it as a reference somewhere - i.e. that transaction is not "stuck" in some singleton. Without the `unwrap()` this situation will go unnoticed.

Or am I reading this wrong?

Reply:
You read that correctly -- yes, that change needs to be removed